### DISCUSSION TITLE
Syntax error compiling CasePath expression with '/' in Swift 5.7

### DISCUSSION THREAD

**Discussion by andreyz** [pointfreeco/swift-case-paths#75] [Q&A] [answered]
<img width="608" alt="image" src="https://user-images.githubusercontent.com/205317/172377402-47eea8ba-502e-471d-b2b3-b5368f19a110.png">

Seems like it's colliding with RegEx literal syntax. ~Any recommended workarounds?~ @stephencelis demonstrates use of initialiser [here in Swift Forums](https://forums.swift.org/t/se-0354-regex-literals/57037/260):
```
-.pullback(state: \Struct.property, action: /Enum.case)
+.pullback(state: \Struct.property, action: CasePath(Enum.case))
```

Here's another workaround, based off [Revised RegEx parsing behaviour](https://github.com/apple/swift/pull/58863):
```
-(\HomeState.route).appending(path: /HomeRoute.game).extract(from:)
+(\HomeState.route).appending(path: (/HomeRoute.game)).extract(from:)
```

Related discussion was in [#72](https://github.com/pointfreeco/swift-case-paths/discussions/72)

PS Seems like neither workarounds are making Xcode 14 previews happy.

**Comment by stephencelis** [comment, marked as answer]
While the new "bare" regex syntax is supposed to be disabled by default, that doesn't appear to be the case in the first beta. Depending on your app's build settings, you should be able to disable it by setting the "Enable Bare Slash Regex Literals" build setting to false.

You could also temporarily use the deprecated `.case(Enum.case)` syntax, but we wouldn't recommend this since it's a lot of churn to contend with for a bug in the first Xcode 14 beta.

Please file feedback, but hopefully this is fixed soon!

Meanwhile, I'm going to convert this to a discussion since it's not a bug in the library, but rather an issue with Xcode.

**Reply by andreyz** [reply]
I'll go with temporarily disabling "bare" regex syntax. Thank you Stephen. I'll post to discussions straight away next time.

**Reply by andreyz** [reply]
Any idea on how to make Swift Previews work even with "Enable Bare Slash Regex Literals" turned to false?

**Reply by stephencelis** [reply]
@andreyz That appears to be a bug in Xcode 14's preview system. Please file feedback if you can! I think a workaround is to move the preview to its own file that does not have any case paths in it.

**Reply by stephencelis** [reply]
FWIW Apple is aware of the issue: https://forums.swift.org/t/se-0354-second-review-regex-literals/57367/100

**Reply by tgrapperon** [reply]
The workaround that consists in adding `//` on the same line as the `CasePath` seems to work (it doesn't fix highlighting, but Previews are building). It seems the best solution in the meantime, especially since we can append some specific string/char to find them back later.
